CHANGELOG.md

```diff
@@ -10,6 +10,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 ### Added
 
 - We added support for searching ShortScience for an entry through the user's browser. [#6018](https://github.com/JabRef/jabref/pull/6018)
+- We updated EditionChecker  to permit edition to start with a number.
```

Comment:
Please add the issue number

Reply:
Nitpick: Please use a single space to separate words in a sentence (after `EditionChecker`, there are two spaces).

Reply:
Hope that custom style formatting is properly working now
Added the issue number and the link as well
Removed one space in a sentence